lib/streamlit/DeltaGenerator.py

```diff
@@ -2014,6 +2037,132 @@ def date_input(self, element, label, value=None, key=None):
         )
         return current_value
 
+    @_with_element
+    def number_input(
+        self,
+        element,
+        label,
+        value=NoValue(),
+        min_value=None,
+        max_value=None,
+        step=None,
+        format=None
+    ):
+        """Display a single-line numeric input widget.
+        Parameters
+        ----------
+        label : str or None
+            A short label explaining to the user what this input is for.
+        value : int/float or None
+            The value of this widget when it first renders.
+            default: min_value or 0
+        min_value : int/float or None
+            The minimum permitted value.
+        max_value : int/float or None
+            The maximum permitted value.
+        step : int/float or None
+            The stepping interval.
+            Defaults to 1 if the value is an int, 0.01 otherwise.
+            If the value is not specified, the format parameter will be used.
+        format : str or None
+            Printf/Python format string.
+        Returns
+        -------
+        int/float
+            The current value of the numeric input widget.
+        Example
+        -------
+        >>> number = st.number_input('Insert a number')
+        >>> st.write('The current number is ', number)
+        """
+
+        from streamlit.util import is_int_value
+
+        if isinstance(value, NoValue):
+            if min_value:
+                value = min_value
+            else:
+                value = 0
+
+        int_value = isinstance(value, int)
+        float_value = isinstance(value, float)
+
+        if value is None:
+            raise ValueError("The value should either be an int/float")
+        else:
+            if format is None:
+                format = "%d" if int_value else "%0.2f"
+
+            if step is None:
+                step = 1 if is_int_value(format % 0.01) else 0.01
```

Comment:
I don't think `step` should depend on `format`.

`format` should only impact the way things appear in the UI.

As for `step`, it should just depend on the data type of `value`.

Reply:
this was meant because the value could be optional in the past

Reply:
Done